src/Numerics/DGMethods/balance_law.jl

```diff
@@ -13,6 +13,45 @@ function calculate_dt(dg, model, Q, Courant_number, t, direction)
     return Courant_number / CFL
 end
 
+using ..Mesh.Filters: FilterIndices
+
+"""
+    FilterStateConservative(model, vars...)
+
+Filter state conservative variables of a given `model`
+based on their names `vars` where `vars` is either a list
+of symbols or strings
+
+## Examples
+```julia
+FiltersStateConservative(model) # applies to the whole state
+FiltersStateConservative(model, :ρe, :(moisture.ρq_tot))
+FiltersStateConservative(model, "ρe", "moisture.ρq_tot")
+```
+"""
+function FilterStateConservative(
+    model,
+    vars... = fieldnames(vars_state_conservative(model, Int))...,
+)
+    # compute indices and create an index target
+    I = Int[]
+    for var in vars
+        if occursin('.', string(var))
+            submodel, var = rsplit(string(var), '.'; limit = 2)
+            vi = varsindex(
+                vars_state_conservative(
+                    getproperty(model, Symbol(submodel)),
+                    Int,
+                ),
+                Symbol(var),
+            )
+        else
+            vi = varsindex(vars_state_conservative(model, Int), Symbol(var))
+        end
+        append!(I, collect(vi))
+    end
```

Comment:
Would this be worth moving into `VariableTemplates`? Something like `varsindices` to compliment `varsindex`?

Also, maybe use `Nothing` instead of `Int`?